pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -176,45 +190,45 @@ func (c StaticResourceController) Sync(ctx context.Context, syncContext factory.
 	}
 
 	errors := []error{}
+	var notFoundErrorsCount int
 	directResourceResults := resourceapply.ApplyDirectly(c.clients, syncContext.Recorder(), c.manifests, c.files...)
 	for _, currResult := range directResourceResults {
+		if apierrors.IsNotFound(currResult.Error) {
```

Comment:
I'm confused; IIUC from our Slack conversation, you wanted this controller to ignore 404 errors coming from the operator's CR. Did you got it wrong?

Reply:
Also, I don't think `ApplyDirectly()` is returning NotFound errors, at least for ServiceMonitors: it calls `ApplyKnownUnstructured()` that calls `ApplyServiceMonitor()`, but that function doesn't return NotFound errors, it tries to create the object:

https://github.com/openshift/library-go/pull/1042/files#diff-55de3da9293d86816615a21ec35c9725ab821c04b938de22d815679554b2fe92R52

Reply:
monitoring-operator creates ServiceMonitor CRD *after* cluster-storage-operator has started. Therefore CSO will try to create ServiceMonitor (CR) when the CRD itself does not exist ye and that's where API server returns 404 (which is quite confusing, btw). And we want to ignore this 404 and not degrade the whole cluster. 1) storage can work well without ServiceMonitor and 2) it will heal when monitoring-operator is started.